src/ScottPlot/plottables/PlottableScatterHighlight.cs

```diff
@@ -0,0 +1,288 @@
+                    float xRight = errorRight.X;
+                    settings.gfxData.DrawLine(penLineError, xLeft, yCenter, xRight, yCenter);
+                    settings.gfxData.DrawLine(penLineError, xLeft, yCenter - errorCapSize, xLeft, yCenter + errorCapSize);
+                    settings.gfxData.DrawLine(penLineError, xRight, yCenter - errorCapSize, xRight, yCenter + errorCapSize);
+                }
+            }
+
+            // draw the lines connecting points
+            if (penLine.Width > 0 && points.Count > 1)
+            {
+                if (stepDisplay)
+                {
+                    List<PointF> pointsStep = new List<PointF>(points.Count * 2);
+                    for (int i = 0; i < points.Count - 1; i++)
+                    {
+                        pointsStep.Add(points[i]);
+                        pointsStep.Add(new PointF(points[i + 1].X, points[i].Y));
+                    }
+                    pointsStep.Add(points[points.Count - 1]);
+                    settings.gfxData.DrawLines(penLine, pointsStep.ToArray());
+                }
+                else
+                {
+                    settings.gfxData.DrawLines(penLine, points.ToArray());
+                }
+            }
+
+            // draw a marker at each point
+            if ((markerSize > 0) && (markerShape != MarkerShape.none))
+                for (int i = 0; i < points.Count; i++)
+                    if (highlightedIndexes.Contains(i))
+                    {
+                        MarkerTools.DrawMarker(settings.gfxData, points[i], highlightedShape, highlightedMarkerSize, highlightedColor);
+                    }
+                    else
+                    {
+                        MarkerTools.DrawMarker(settings.gfxData, points[i], markerShape, markerSize, color);
+                    }
+
+        }
+
+        public void SaveCSV(string filePath, string delimiter = ", ", string separator = "\n")
+        {
+            System.IO.File.WriteAllText(filePath, GetCSV(delimiter, separator));
+        }
+
+        public string GetCSV(string delimiter = ", ", string separator = "\n")
+        {
+            StringBuilder csv = new StringBuilder();
+            for (int i = 0; i < ys.Length; i++)
+                csv.AppendFormat("{0}{1}{2}{3}", xs[i], delimiter, ys[i], separator);
+            return csv.ToString();
+        }
+
+        public void HighlightClear()
+        {
+            highlightedIndexes.Clear();
+        }
+
+        public void HighlightPoint(int index)
+        {
+            highlightedIndexes.Add(index);
+        }
+
+        public void HighlightPointNearest(double x)
+        {
+            int index = xs.Select((p, i) => (p, i)).OrderBy(p => Math.Abs(p.p - x)).First().i;
+            HighlightPoint(index);
+        }
+
+        public void HighlightPointNearest(double x, double y)
+        {
+            List<(double x, double y)> points = xs.Zip(ys, (first, second) => (first, second)).ToList();
```

Comment:
Since I changed it to a for loop you do. You cannot index an IEnumerable, you can only enumerate it. Since the index was important I kept the `.ToList()`